### ISSUE TITLE
attachments (images) not updating when attachment file changes

### ISSUE THREAD

**Issue by jdbrice** [fabiospampinato/notable#301]
- **OS Version**: MacOs High Sierra 10.13.6
- **Notable Version**: 1.1.0

### Current behavior

Image attachment copied into attachment directory
note has a ref to image via:
```html
<img src="@attachment/image_name.png" width="100px" />
```
change image in the attachment directory (copy new image with the same name)

Inside Notable, the image does not update after toggling edit mode/ preview mode or changing to another note and then coming back

### Expected behavior

Image preview inside note should update automatically with file change or at least after a manual toggle edit/preview mode. Or by leaving note and coming back to it. I looked at a few related issues but I don't think any others quite cover this behavior, but sorry if I missed it.

Thanks for working on Notable, it is a very useful 😄

**Comment by jdbrice** [comment]
I guess it is due to some kind of caching? I can imagine that implementing a solution to watch attachments would be tricky. Without implementing a full-fledged automatic solution would it be possible to at least do a full refresh when switching to another note and back or switching to edit mode and back to preview?  That would already improve the issue/workflow for me quite a bit.

**Comment by fabiospampinato** [comment]
> I guess it is due to some kind of caching? 

Mostly we would have to read each note and check which of them are embedding which attachments, then if any of those attachments change (we are already listening for this) we would have to update those notes. 

> Without implementing a full-fledged automatic solution would it be possible to at least do a full refresh when switching to another note and back or switching to edit mode and back to preview?

The note should already get re-rendered when switching to edit mode and then to preview mode again, is this not happening? 🤔 Are you saying this because an attachment that no longer exists is rendered properly?

**Comment by jdbrice** [comment]
> The note should already get re-rendered when switching to edit mode and then to preview mode again

This does not produce the expected result for me. I can delete the attachment and the note will continue to render the image (but the link is broke, i.e. does nothing). I can toggle edit/preview or go to another note and it does not properly refresh.  Only after exiting Notable and restarting the program does the note render correctly. 

The only exception to this is the other way around. If the attachment does not exist when first rendered, then when the attachment is added the render will refresh after a toggle of edit->preview. But then further changes to the file (including deleting) are not observed within Notable.

Specifically my workflow involves generating plots with some other analysis software which updates the images with the same name in the Notable attachment directory. which BTW, I really like being able to use markdown + filesystem for attachments, it is a great combo.

**Comment by fabiospampinato** [comment]
> This does not produce the expected result for me.

Can you open a separate issue about this? I think Electron might be doing some caching here or something.

> which BTW, I really like being able to use markdown + filesystem for attachments, it is a great combo.

Agree, it can be pretty powerful to have attachments freely available from the filesystem.

**Comment by fabiospampinato** [comment]
This issue could be resolved by switching to UUIDs #538
